frontend/src/components/LaunchForm.tsx

```diff
@@ -0,0 +1,131 @@
+import {DEFUALT_ADDRESS, DEFAULT_PORT} from "../consts"
+
+interface LaunchFormProps {
+    adress : string;
+    setAddress : (s : string) => void;
+    port : string;
+    setPort : (s : string) => void;
+    login : string, 
+    setLogin : (s : string) => void;
+    password : string;
+    setPassword : (s : string) => void;
+    setFormPassed : (b : boolean) => void;
+}
+
+function LaunchForm({adress, setAddress, port, setPort, login, setLogin, password, setPassword, setFormPassed} : LaunchFormProps)
+{
+    const classes = useStyles();
+
+    const changeAddress =  (event : React.ChangeEvent<HTMLInputElement>) => {
+        setAddress(event.target.value.length && event.target.value[0].trim() === '' ? 
+            event.target.value.slice(1) : event.target.value);
+    }
+
+    const changePort = (event : React.ChangeEvent<HTMLInputElement>)  => {
+        setPort(event.target.value.length && event.target.value[0].trim() === '' ? 
+            event.target.value.slice(1) : event.target.value);
+    }
+
+    const changeLogin =  (event : React.ChangeEvent<HTMLInputElement>) => {
+        setLogin(event.target.value.length && event.target.value[0].trim() === '' ? 
+            event.target.value.slice(1) : event.target.value);
+    }
+
+    const changePassword = (event : React.ChangeEvent<HTMLInputElement>)  => {
+        setPassword(event.target.value.length && event.target.value[0].trim() === '' ? 
+            event.target.value.slice(1) : event.target.value);
+    }
+
+    const submitForm = (e : any) => {
+        e.preventDefault();
+
+        if (adress === "")
+            setAddress(DEFUALT_ADDRESS);
+
+        if (port == "")
+            setPort(DEFAULT_PORT);
+
+        //console.log(adress, port, login, password)
+        setFormPassed(true);
+    }
+
+    return <div className={classes.blurredBackground}>
+        <form className={classes.content} onSubmit={submitForm} >
+            <div className={classes.element}>
+                <div>Address:</div>
+                <input
+                    defaultValue=""
+                    value={adress}
+                    onChange={changeAddress}
+                    placeholder={DEFUALT_ADDRESS}
+                />
+            </div>
+
+            <div className={classes.element}>
+                <div>Port:</div>
+                <input
+                    defaultValue=""
+                    value={port}
+                    onChange={changePort}
+                />
+            </div>
+
+            <div className={classes.element}>
+                <div>Username:</div>
+                <input
+                    defaultValue=""
+                    value={login}
+                    onChange={changeLogin}
+                />
+            </div>
+
+            <div className={classes.element}>
+                <div>Password:</div>
+                <input
+                    defaultValue=""
+                    value={password}
```

Comment:
<input type="password" value="2137">
dobra 
nie wiem czy to zgodne ze sztuką xd

Reply:
?